### PR TITLE
Adding generate_alert to create/update indicator.

### PR DESCRIPTION
**Pull request by ANT1OFF**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Description
Adding generate_alert as an option to the create/update indicator in security center command. Setting the default to True to keep the same behaviour as before. This is primarily useful when using the command to add indicators as allowed.

Adding the generate_alert command to the false positive playbook and setting it to default to false.

## Screenshots
image
image

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

### CODE REVIEW

**Comment by ANT1OFF** [comment]
The changes made to the playbook are minimal to get it working and not generate future alerts on allowed indicators. The input configuration should probably be propagated to the other MDE I&R playbooks. Alternatively, the False Positive playbook could prompt the user to select whether to generate an alert when a given indicator is added to as allowed.  @ssokolovich any thoughts on what approach would be best?

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py [hunk 1]
```diff
@@ -3873,12 +3875,13 @@ def sc_create_update_indicator_command(client: MsClient, args: Dict[str, str]) -
     indicator_application = args.get('indicator_application', '')
     recommended_actions = args.get('recommended_actions', '')
     rbac_group_names = argToList(args.get('rbac_group_names', []))
+    generate_alert = args.get('generate_alert', True)
```
Comment (YuvHayun) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Integrations/MicrosoftDefenderAdvancedThreatProtection/MicrosoftDefenderAdvancedThreatProtection.py:3878]:
```suggestion
    generate_alert = argToBoolean(args.get('generate_alert', True))
```

**Review by YuvHayun — CHANGES_REQUESTED** [review]
Hi @ANT1OFF, thx for your contribution!
I added a small fix request.
Also:
1. Please add release notes, you can refer to [this guide](https://xsoar.pan.dev/docs/documentation/release-notes).
2. I requested @idovandijk to review the playbook changes so wait for further instructions.

Feel free to contact me if you have any questions.
I'm available here and on slack.

File: Packs/MicrosoftDefenderAdvancedThreatProtection/ReleaseNotes/1_12_3.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+
+#### Integrations
+##### Microsoft Defender for Endpoint
+- Adding the option to disable the generate alert flag when adding an indicator to Microsoft Security Center. Primarily useful when setting indicators to allow.
+
+#### Playbooks
+##### MDE - False Positive Incident Handling
+- Added the ability to set indicator alert generation behaviour as a playbook option.
+- Updated the playbook to set false positive indicators not to generate an alert by default.
```
Comment (YuvHayun) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/ReleaseNotes/1_12_3.md:9]:
```suggestion
- Improved implementation of the playbook to set false positive indicators not to generate an alert by default.
```

File: Packs/MicrosoftDefenderAdvancedThreatProtection/ReleaseNotes/1_12_3.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
+
+#### Integrations
+##### Microsoft Defender for Endpoint
+- Adding the option to disable the generate alert flag when adding an indicator to Microsoft Security Center. Primarily useful when setting indicators to allow.
```
Comment (YuvHayun) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/ReleaseNotes/1_12_3.md:4]:
```suggestion
- Added the option to disable the generate alert flag when adding an indicator to Microsoft Security Center. Primarily useful when setting indicators to allow.
```

**Review by YuvHayun — CHANGES_REQUESTED** [review]
@ANT1OFF thx for the quick reply!
small fix regarding the RN.

**Comment by YuvHayun** [comment]
@ANT1OFF thx again for the quick reply!
Let's wait for Ido's review.

**Comment by ssokolovich** [comment]
@ANT1OFF - Awesome enhancement! 
As for your question, I think we should give that as an option through the data collection task and not the input (note that you haven't added this input to the PB inputs).
Should look like this ( so we can give the decision to the analyst and not something more hardcoded such as PB input):
image
In this way, the default is always false (to avoid mistakes) - and the input was always provided to the user.  Let me know your thoughts on this :)

**Comment by ANT1OFF** [comment]
@ssokolovich agreed the second option would be best :)

File: Packs/MicrosoftDefenderAdvancedThreatProtection/Playbooks/playbook-MDE_-_False_Positive_Incident_Handling_README.md [hunk 4]
```diff
@@ -33,7 +33,7 @@ SearchIncidentsV2
 | CloseDuplicate | Whether the duplicate incidents should be closed as well in the Microsoft Defender for Endpoint instance.<br/>The playbook looks for the world "Close" in this input.. |  | Optional |
 | HostID | The ID of the host for running an un-isolation process. | ${incident.deviceid} | Optional |
 | FileSha256 | Enter the File SHA256 you would like to block. | ${incident.filesha256} | Optional |
-
+| GenerateAlert | Whether an alert should be created for future occurrences of the indicator. | False | Optional |
```
Comment (ssokolovich) [inline Packs/MicrosoftDefenderAdvancedThreatProtection/Playbooks/playbook-MDE_-_False_Positive_Incident_Handling_README.md:36]: @YuvHayun - no need for this input.

**Review by YuvHayun — APPROVED** [review]

**Pull request merged**